src/query/server/query.go

```diff
@@ -126,8 +126,14 @@ type RunOptions struct {
 
 	// CustomPromQLParseFunction is a custom PromQL parsing function.
 	CustomPromQLParseFunction promql.ParseFn
+
+	// ApplyCustomTSDBOptions is a transform that allows for custom tsdb options.
+	ApplyCustomTSDBOptions CustomTSDBOptionsFn
```

Comment:
Maybe name this `CustomTSDBOptionsFunction` so the name is similar to the above field?

Reply:
Yeah was thinking that, but this require an application since it's a transform function against the tsdb options

Reply:
Yeah makes sense